docs/architecture/porting-existing-aspnet-apps/more-migration-scenarios.md

```diff
@@ -112,10 +112,415 @@ public IActionResult Index()
+{
+    public override void OnActionExecuting(ActionExecutingContext filterContext)
+    {
+        filterContext.RequestContext.HttpContext.Response.AddHeader(
+            "Access-Control-Allow-Origin", "example.com");
+        base.OnActionExecuting(filterContext);
+    }
+}
+```
+
+ASP.NET Web API can also use such a filter, but it has [built-in support for enabling CORS](/aspnet/web-api/overview/security/enabling-cross-origin-requests-in-web-api#enable-cors) as well:
+
+```csharp
+public static class WebApiConfig
+{
+    public static void Register(HttpConfiguration config)
+    {
+        config.EnableCors();
+        // ...
+    }
+}
+```
+
+Once this is added, you can configure allowed origins, headers, and methods using the `EnableCors` attribute, like so:
+
+```csharp
+[EnableCors(origins: "https://dot.net", headers: "*", methods: "*")]
+public class TestController : ApiController
+{
+    // Controller methods not shown...
+}
+```
+
+Before migrating your CORS implementation from ASP.NET MVC 5 or ASP.NET Web API 2, be sure to review [how CORS works](/aspnet/web-api/overview/security/enabling-cross-origin-requests-in-web-api#how-cors-works) and create some automated tests that demonstrate CORS is working as expected in your current system.
+
+In ASP.NET Core, there are three built-in ways to enable CORS:
+
+- [Configured via policy](/aspnet/core/security/cors?#cors-with-named-policy-and-middleware) in `ConfigureServices`
+- Enabled with [endpoint routing](/aspnet/core/security/cors?#enable-cors-with-endpoint-routing)
+- Enabled with the [`EnableCors` attribute](/aspnet/core/security/cors?view=aspnetcore-5.0#enable-cors-with-attributes)
+
+Each of these approaches is covered in detail in the docs, which are linked from the above options. Which one you choose will largely depend on how your existing app supports CORS. If the app uses attributes, you can probably migrate to use the `EnableCors` attribute most easily. If your app uses filters, you could continue using that approach (though it's not the typical approach used in ASP.NET Core), or migrate to use attributes or policies. Endpoint routing is a relatively new feature introduced with ASP.NET Core 3 and as such it doesn't have a close analog in ASP.NET MVC 5 or ASP.NET Web API 2 apps.
+
+## Custom areas
+
+Many ASP.NET MVC apps use Areas to organize the project. Areas typically reside in the root of the project in an *Areas* folder, and must be registered when the application starts, typically in `Application_Start()`:
+
+```csharp
+AreaRegistration.RegisterAllAreas();
+```
+
+An alternative to registering all areas in startup is to use the `RouteArea` attribute on individual controllers:
+
+```csharp
+[RouteArea("Admin")]
+public class SomeController : Controller
+```
+
+When using Areas, additional arguments are passed into HTML helper methods to generate links to actions in different areas:
+
+```html
+@Html.ActionLink("News", "Index", "News", new { area = "News" }, null)
+```
```

Comment:
Doesn't seem the suitable language identifier

Reply:
Is there a better one for Razor?

Reply:
I think cshtml may work well.

Reply:
Ok, thanks. Found this: https://docs.microsoft.com/en-us/contribute/code-in-docs#supported-languages and agree. Will fix.